response_operations_ui/views/messages.py

```diff
@@ -195,6 +199,38 @@ def view_selected_survey(selected_survey):
                                response_error=True)
 
 
+@messages_bp.route('/close-conversation', methods=['GET', 'POST'])
+@login_required
+def close_conversation():
+    try:
+        messages = session['messages']
+        selected_survey = session['messages_survey_selection']
+        thread_id = messages[0]['thread_id']
```

Comment:
We shouldn't be using the session to store state across sequential requests like this. If you put `close-conversation` on `/threads/<thread_id>` - `/threads/<thread_id>/close-conversation` you'd have access to the thread id for the conversation being closed.